### PR TITLE
shell_commands: provide command for 6Lo context administration at 6LBR

### PR DESCRIPTION
**Pull request by authmillenon**
Provides a shell command to administrate 6LoWPAN compression contexts at the border router.

Depends on ~~#3134~~ (merged) (and #3525 to work properly).

### CODE REVIEW

**Comment by authmillenon** [comment]
Rebased to current #3134

**Comment by authmillenon** [comment]
Rebased to current #3134.

**Comment by authmillenon** [comment]
Rebased to current #3134

**Comment by authmillenon** [comment]
Rebased to current #3134

**Comment by authmillenon** [comment]
Rebased to current #3134

**Comment by authmillenon** [comment]
Rebased to current master

**Comment by OlegHahm** [comment]
What do I have to do to get this building with `gnrc_networking` example? Just replacing `gnrc_ipv6_router_default` by `gnrc_sixlowpan_nd_border_router` or adding it doesn't seem to work.

**Comment by authmillenon** [comment]
`gnrc_ipv6_border_router_default`

**Comment by OlegHahm** [comment]
make sense ;)

**Comment by OlegHahm** [comment]
For the record: the (pseudo) module is called `gnrc_sixlowpan_border_router_default`.

A dependency to `xtimer` is missing.

File: sys/shell/commands/sc_gnrc_6ctx.c [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    gnrc_sixlowpan_nd_router_abr_rem_ctx(abr, cid);
+    del_timer[cid].callback = NULL;
+}
+
+static void _usage(char *cmd_str)
+{
+    printf("usage: %s [{add <0-15> <prefix>/<prefix_len> <ltime in min>|del <ctx>}]\n", cmd_str);
+    puts("       Note that `del` only will invalidate the context for compression.");
+    puts("       This way the network has time to get the context update dissiminated.");
+    puts("       It will be reassignable after 5 min.");
```
Comment (OlegHahm) [inline sys/shell/commands/sc_gnrc_6ctx.c]: I think you can shorten this to something like: "'del' will only invalidate the context for compression which can be reassigned after 5 min."

**Comment by authmillenon** [comment]
> A dependency to xtimer is missing.

Should I add it as a dependency for `shell_commands`? That seems weird to me... Because the `xtimer` is only used by the shell command, not the border router it self (yet).

**Comment by OlegHahm** [comment]
Hm, maybe we just wait for switch to xtimer then...

**Comment by kaspar030** [comment]
+1

**Comment by authmillenon** [comment]
As stated in the description ;-)

**Comment by OlegHahm** [comment]
I guess a rebase is required.

**Comment by authmillenon** [comment]
Rebased

File: sys/shell/commands/sc_gnrc_6ctx.c (hunk 1 above)
Comment (OlegHahm) [inline sys/shell/commands/sc_gnrc_6ctx.c]: .

Comment (authmillenon) [inline sys/shell/commands/sc_gnrc_6ctx.c]: addressed

**Comment by OlegHahm** [comment]
please squash

**Comment by authmillenon** [comment]
Squashed

**Comment by authmillenon** [comment]
(but not pushed >.<)

**Comment by OlegHahm** [comment]
How can I test and how and when should the configured context get propagated?

**Comment by OlegHahm** [comment]
After configuring a context for my prefix, pinging to or via the 6LBR doesn't work any more.

**Comment by OlegHahm** [comment]
However, that's probably not related to the shell commands themselves.

**Comment by authmillenon** [comment]
Rebased to current master.

**Comment by authmillenon** [comment]
Added immediate advertisement for contexts (and fixed a bug that was revealed by that in `net/gnrc/ndp/internal.h`)

**Comment by authmillenon** [comment]
Fix for https://github.com/RIOT-OS/RIOT/pull/3750#issuecomment-141579928 will follow. I think I know what's happening.

**Comment by authmillenon** [comment]
> Fix for #3750 (comment) will follow. I think I know what's happening.

it's #3928 if that wasn't clear.

**Comment by OlegHahm** [comment]
ACK

**Comment by OlegHahm** [comment]
Please squash and merge as soon as Travis is green. With #3949 this is working. (However, this PR iteelf is sane anyway.)

**Comment by authmillenon** [comment]
Squashed and rebased.
